test/services/geolocation_service_test.dart

```diff
@@ -112,5 +119,95 @@ void main() {
       verify(mockGeolocator.requestPermission()).called(1);
       expect(actualGeoPoint, equals(expectedGeoPoint));
     });
+
+    test("determinePosition returns a stream", () async {
+      const latitude = 37.4219983;
+      const longitude = -122.084;
+
+      when(mockGeolocator.isLocationServiceEnabled())
+          .thenAnswer((_) async => true);
+      when(mockGeolocator.checkPermission())
+          .thenAnswer((_) async => LocationPermission.always);
+      when(
+        mockGeolocator.getCurrentPosition(
+          locationSettings: geoLocationService.locationSettings,
+        ),
+      ).thenAnswer(
+        (_) async => getSimplePosition(latitude, longitude),
+      );
+      when(
+        mockGeolocator.getPositionStream(
+          locationSettings: geoLocationService.locationSettings,
+        ),
+      ).thenAnswer(
+        (_) => Stream.fromIterable([getSimplePosition(latitude, longitude)]),
+      );
+
+      final streamController = StreamController<GeoPoint?>();
+      await liveGeoLocationService.determinePosition(
+        streamController,
+      );
+
+      final actualGeoPoint = await streamController.stream.first;
+
+      expect(actualGeoPoint, equals(const GeoPoint(latitude, longitude)));
```

Comment:
I think it would make sense to test more than one value in this test, as this services outputs a stream of multiple positions.

Reply:
Fixed in 7241a30